disnake/poll.py

```diff
@@ -0,0 +1,423 @@
+# SPDX-License-Identifier: MIT
+
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import TYPE_CHECKING, Dict, List, Optional, Union
+
+from . import utils
+from .abc import Snowflake
+from .emoji import Emoji, _EmojiTag
+from .enums import PollLayoutType, try_enum
+from .iterators import PollAnswerIterator
+from .partial_emoji import PartialEmoji
+
+if TYPE_CHECKING:
+    from datetime import datetime
+
+    from .message import Message
+    from .state import ConnectionState
+    from .types.poll import (
+        Poll as PollPayload,
+        PollAnswer as PollAnswerPayload,
+        PollCreateAnswerPayload,
+        PollCreateMediaPayload,
+        PollCreatePayload,
+        PollMedia as PollMediaPayload,
+    )
+
+__all__ = (
+    "PollMedia",
+    "PollAnswer",
+    "Poll",
+)
+
+
+class PollMedia:
+    """Represents data of a poll's question/answers.
+
+    .. versionadded:: 2.10
+
+    Parameters
+    ----------
+    text: :class:`str`
+        The text of this media.
+    emoji: Optional[Union[:class:`Emoji`, :class:`PartialEmoji`, :class:`str`]]
+        The emoji of this media.
+
+    Attributes
+    ----------
+    text: Optional[:class:`str`]
+        The text of this media.
+    emoji: Optional[:class:`PartialEmoji`]
+        The emoji of this media.
+    """
+
+    __slots__ = ("text", "emoji")
+
+    def __init__(
+        self, text: Optional[str], *, emoji: Optional[Union[Emoji, PartialEmoji, str]] = None
+    ) -> None:
+        if text is None and emoji is None:
+            raise ValueError("At least one of `text` or `emoji` must not be None")
+
+        self.text = text
+        self.emoji: Optional[Union[Emoji, PartialEmoji]] = None
+        if isinstance(emoji, str):
+            self.emoji = PartialEmoji.from_str(emoji)
+        elif isinstance(emoji, _EmojiTag):
+            self.emoji = emoji
+        else:
+            if emoji is not None:
+                raise TypeError("Emoji must be None, a str, PartialEmoji, or Emoji instance.")
```

Comment:
Now i remember because there was that useless check for a None value, it's because here i'm validating the type at runtime, should i drop this?

Reply:
it's fine like this, silently ignoring wrong input types usually isn't good UX